extensions/cms/src/test/cmsResource/tree/treeProvider.test.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as should from 'should';
+import * as TypeMoq from 'typemoq';
+import 'mocha';
+import { AppContext } from '../../../appContext';
+import { ApiWrapper } from '../../../apiWrapper';
+
+import { CmsResourceTreeProvider } from '../../../cmsResource/tree/treeProvider';
+import { CmsResourceMessageTreeNode } from '../../../cmsResource/messageTreeNode';
+import { CmsResourceEmptyTreeNode } from '../../../cmsResource/tree/cmsResourceEmptyTreeNode';
+import { CmsResourceTreeNode } from '../../../cmsResource/tree/cmsResourceTreeNode';
+
+// Mock services
+let mockAppContext: AppContext;
+let mockExtensionContext: TypeMoq.IMock<vscode.ExtensionContext>;
+let mockApiWrapper: TypeMoq.IMock<ApiWrapper>;
+
+
+describe('CmsResourceTreeProvider.getChildren', function (): void {
```

Comment:
Suggest also adding tests to validate that after adding children the provider returns them correctly